properties/pkg/properties_test.go

```diff
@@ -0,0 +1,384 @@
+// Copyright © 2021 Banzai Cloud
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package properties
+
+import (
+	"encoding/json"
+	"fmt"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+)
+
+func TestNewProperties(t *testing.T) {
+	p := NewProperties()
+
+	if len(p.keys) != 0 && len(p.properties) != 0 {
+		t.Errorf("Newly created Properties is not empty")
+	}
+}
+
+func TestProperties_Get(t *testing.T) {
+	k := "test.key"
+	v := "test.value"
+	c := "this is a comment"
+
+	p := NewProperties()
+	_ = p.Set(k, v, c)
+
+	t.Run("Get existing Property", func(t *testing.T) {
+		if _, found := p.Get(k); !found {
+			t.Errorf("Existing Property is not found. Expected: %v, got %v", found, !found)
+		}
+	})
+
+	t.Run("Get nonexistent Property", func(t *testing.T) {
+		if _, found := p.Get(k + "-"); found {
+			t.Errorf("None existing Property found. Expected: %v, got %v", !found, found)
+		}
+	})
+
+	t.Run("Expected Property is returned", func(t *testing.T) {
+		expectedProp := Property{
+			key:     k,
+			value:   v,
+			comment: c,
+		}
+
+		if prop, _ := p.Get(k); !prop.Equal(expectedProp) {
+			t.Errorf("Mismatch between the returned and the expected Property")
+		}
+	})
+}
+
+func TestProperties_Keys(t *testing.T) {
+
+	p := NewProperties()
+	_ = p.Set("test.key", "test.value", "this is a comment line")
```

Comment:
I'd also assert whether the `error` return by `Set` is not nil. This is true for every occurrence of this all along this test.